tests/main/failover/task.yaml

```diff
@@ -95,8 +95,11 @@ execute: |
             exit 1
         fi
 
-        # install new target snap
-        snap install --dangerous failing.snap
+        # Retry installing new target snap until it is intalled correctly. The installation
+        # could fail with: cannot finish core installation, there was a rollback across reboot
+        while ! snap install --dangerous failing.snap; do
```

Comment:
Isn't this hiding the real problem we might have?

Reply:
@stolowski that could be true, initially I though there was a problem with the snap that we are modifying, so I created this workaround. 
Could you confirm that the solution you are implementing would fix this problem too?

Reply:
No, my fix is unrelated to this problem. However it seems this problem of rollback across reboot is now understood and might have a fix soon, @mvo5 has more details.

Reply:
We have fixed a bug about the "rollback accross reboots" just recently - its in edge now. It would be great to check (cc @sergiocazzolato) if this bug is still reproducible with the "edge" snap. There is one more PR in flight (#7020) that may also help/fix this issue.

Reply:
@mvo5 sure, will test that